From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Thu, 29 Oct 2026 15:55:40 +0100
Subject: [PATCH] adb: listen on USB only

adbd listened on TCP port 5555, on every interface, whenever it found no
USB gadget, and on any port named by service.adb.tcp.port or
persist.adb.tcp.port. LuneOS has no Android properties to set those
with, and on x86 property_get() asks a socket in /tmp that any local user
can create. adbd now serves USB only.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Inappropriate [LuneOS-specific: adb over USB only]
---
 adb/adb.c | 30 +++++++++---------------------
 1 file changed, 9 insertions(+), 21 deletions(-)

diff --git a/adb/adb.c b/adb/adb.c
index 0cfc982136..7d4f3504d5 100644
--- a/adb/adb.c
+++ b/adb/adb.c
@@ -1302,9 +1302,6 @@ static int should_drop_privileges() {
 int adb_main(int is_daemon, int server_port)
 {
 #if !ADB_HOST
-    int port;
-    char value[PROPERTY_VALUE_MAX];
-
     umask(000);
 #endif
 
@@ -1402,27 +1399,18 @@ int adb_main(int is_daemon, int server_port)
         }
     }
 
-    int usb = 0;
+    /*
+     * LuneOS: USB only. adbd used to listen on TCP port 5555, on every
+     * interface, whenever it found no USB gadget, and on the port in
+     * service.adb.tcp.port or persist.adb.tcp.port. There are no Android
+     * properties here to set those, and on x86 property_get() asks a socket
+     * in /tmp that any local user can make (see auth_enabled above).
+     */
     if (access(USB_ADB_PATH, F_OK) == 0 || access(USB_FFS_ADB_EP0, F_OK) == 0) {
         // listen on USB
         usb_init();
-        usb = 1;
-    }
-
-    // If one of these properties is set, also listen on that port
-    // If one of the properties isn't set and we couldn't listen on usb,
-    // listen on the default port.
-    property_get("service.adb.tcp.port", value, "");
-    if (!value[0]) {
-        property_get("persist.adb.tcp.port", value, "");
-    }
-    if (sscanf(value, "%d", &port) == 1 && port > 0) {
-        printf("using port=%d\n", port);
-        // listen on TCP port specified by service.adb.tcp.port property
-        local_init(port);
-    } else if (!usb) {
-        // listen on default port
-        local_init(DEFAULT_ADB_LOCAL_TRANSPORT_PORT);
+    } else {
+        D("No USB gadget for adb; adbd listens on nothing else\n");
     }
 
     D("adb_main(): pre init_jdwp()\n");
-- 
2.53.0

